### PR TITLE
Fix InferenceAPI on image task

### PR DESCRIPTION
**Pull request by Wauplin**
Related to [slack discussion](https://huggingface.slack.com/archives/C02V5EA0A95/p1671111270868539) (internal link).

- `text-to-image` now works by returning a Pillow image (if PIL is installed, otherwise you get an `ImportError`).
- if content type is `application/json` or `image/jpeg` => we parse the output. Otherwise raise `NotImplementedError`. This is in particular the case for `text-to-speech` tasks.
- the InferenceAPI has now a kwarg `raw_response: bool = False`. If `True`, we return the full `Response` object to the user. At least this way if we don't handle a task yet, the user can build its own parser (typically for audio).
- adapted/fixed some stuff (now sending user agent in request for example)
- => added 2 examples[ in the doc](https://moon-ci-docs.huggingface.co/docs/huggingface_hub/pr_1270/en/package_reference/inference_api) and some tests

### CODE REVIEW

File: src/huggingface_hub/inference_api.py [hunk 1]
```diff
@@ -136,28 +145,56 @@ def __init__(
...
+                raise ImportError(
+                    f"Task '{self.task}' returned as image but Pillow is not installed."
+                    " Please install it (`pip install Pillow`) or pass"
+                    " `raw_response=True` to get the raw `Response` object and parse"
+                    " the image by yourself."
+                )
+
+            from PIL import Image
+
+            return Image.open(io.BytesIO(response.content))
```
Comment (osanseviero) [inline src/huggingface_hub/inference_api.py:192]: Very cool!

File: src/huggingface_hub/inference_api.py [hunk 2]
```diff
@@ -136,28 +145,56 @@ def __init__(
...
+            f" options={self.options})"
+        )
 
     def __call__(
         self,
         inputs: Optional[Union[str, Dict, List[str], List[List[str]]]] = None,
         params: Optional[Dict] = None,
         data: Optional[bytes] = None,
-    ):
+        raw_response: bool = False,
```
Comment (osanseviero) [inline src/huggingface_hub/inference_api.py:159]: Maybe this is a good opportunity to add some docstring for this and its arguments

**Review by osanseviero — APPROVED** [review]
LGTM!

File: src/huggingface_hub/inference_api.py (hunk 2 above)
Comment (Wauplin) [reply src/huggingface_hub/inference_api.py:159]: I felt when doing so that I was lazy here :innocent:

Comment (Wauplin) [reply src/huggingface_hub/inference_api.py:159]: Done in https://github.com/huggingface/huggingface_hub/pull/1270/commits/7ceef22e15fa4ff439168ad87535d302476d382b.

**Comment by Wauplin** [comment]
Thanks for the review @osanseviero ! I've added the docstring and will now merge :)

**Pull request merged**